src/util/migrate.cpp

```diff
@@ -510,9 +511,10 @@ expr2tc sym_name_to_symbol(irep_idt init, type2tc type)
 
     char *endatptr, *endexmptr;
     level1_num = strtol(atstr.c_str(), &endatptr, 10);
+    // assertion could fail due to invalid symbol identifier.
     assert(endatptr != atstr.c_str());
     thread_num = strtol(exmstr.c_str(), &endexmptr, 10);
-    assert(endexmptr != exmstr.c_str());
+    assert(endatptr != atstr.c_str());
```

Comment:
Please restore the original check here. No use checking the same thing twice ;)

Reply:
Good spot! 
sorry, it's a copy-paste error...